service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -0,0 +1,206 @@
+                    .map(value -> extractDevice(value))
+                    .map(text -> REQUESTER_DEVICE + text)
+                    .orElseThrow(() -> new EhrMapperException("Could not resolve Device Reference"));
+            } else if (reference.getResourceType().equals(ResourceType.Organization.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (Organization) resource)
+                    .map(value -> REQUESTER_ORG + value.getName())
+                    .orElseThrow(() -> new EhrMapperException("Could not resolve Organization Reference"));
+            } else if (reference.getResourceType().equals(ResourceType.Patient.name())) {
+                return REQUESTER_PATIENT;
+            } else if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
+                return messageContext.getInputBundleHolder()
+                    .getResource(reference)
+                    .map(resource -> (RelatedPerson) resource)
+                    .map(value -> REQUESTER_RELATION + extractHumanName(value.getNameFirstRep()))
+                    .orElseThrow(() -> new EhrMapperException("Could not resolve RelatedPerson Reference"));
+            } else if (reference.getResourceType().equals(ResourceType.Practitioner.name())) {
+                return StringUtils.EMPTY;
+            }
+            throw new EhrMapperException("Requester Reference not of expected Resource Type");
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private String buildSpecialtyDescription(ReferralRequest referralRequest) {
+        Optional<String> specialty = Optional.empty();
+        if (referralRequest.hasSpecialty()) {
+            CodeableConcept specialtyObject = referralRequest.getSpecialty();
+            specialty = extractTextOrCoding(specialtyObject);
+        }
+        return specialty.map(value -> SPECIALTY + value).orElse(StringUtils.EMPTY);
+    }
+
+    private String buildRecipientDescription(ReferralRequest referralRequest) {
+        if (referralRequest.hasRecipient()) {
+            var ignoreFirstPractitioner = referralRequest.getRecipient();
+            removeFirstPractitionerReference(ignoreFirstPractitioner);
+
+            return ignoreFirstPractitioner.stream()
+                .filter(Reference::hasReferenceElement)
+                .map(value -> extractRecipient(messageContext, value))
+                .collect(Collectors.joining(COMMA));
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private void removeFirstPractitionerReference(List<Reference> referenceList) {
+        for (int index = 0; index < referenceList.size() - 1; index++) {
+            if (referenceList.get(index).getReference().startsWith(ResourceType.Practitioner.name())) {
+                referenceList.remove(index);
+                return;
+            }
+        }
+    }
```

Comment:
This code still removes an item from the underlying referralRequest.getRecipient() list. You need to create an entirely new lines that only contains the objects you need. One way to do this is by copying just those objects to the new list.

```suggestion
private String buildRecipientDescription(ReferralRequest referralRequest) {
        if (referralRequest.hasRecipient()) {
            return getRecipientsWithoutFirstPractitioner(referralRequest).stream()
                .filter(Reference::hasReferenceElement)
                .map(value -> extractRecipient(messageContext, value))
                .collect(Collectors.joining(COMMA));
        }

        return StringUtils.EMPTY;
    }

    private List<Reference> getRecipientsWithoutFirstPractitioner(ReferralRequest referralRequest) {
        boolean firstPractitionerFound = false;
        List<Reference> recipients = new ArrayList<>(referralRequest.getRecipient().size());
        for (Reference reference : referralRequest.getRecipient()) {
            if (!firstPractitionerFound && isReferenceToPractitioner(reference)) {
                firstPractitionerFound = true;
            } else {
                recipients.add(reference);
            }
        }
        return recipients;
    }

    private boolean isReferenceToPractitioner(Reference reference) {
        return reference.getReference().startsWith(ResourceType.Practitioner.name());
    }
```